splunk_topology/test_splunk_topology.py

```diff
@@ -1237,3 +1326,210 @@ def _mocked_extract_components(*args, **kwargs):
         # check if the check continued and finished
         self.assertEqual(instance[0]["stop_snapshot"], True)
         self.assertEqual(instance[0]["start_snapshot"], True)
+
+
+class TestSplunkTokenBasedAuth(AgentCheckTest):
```

Comment:
1. test Exception raised when you forget to set a mandatory value
2. test that token_auth is preferred over basic_auth? set values for both and token based should be used.
3. test that a token is renewed when it's almost expired
4. test exception when a token could not be generated, say a connectivity error to Splunk
5. test in where the memory token is expired